patch: add reference and explanation pages (#116)
canonical/mongo-single-kernel-library: docs/explanation/security/cryptography.md

2. Client authentication to MongoDB

### MongoDB replica/shard/router communication (i.e. Internal Membership) 

Internal Members authenticate using:
1. KeyFiles using SCRAM-SHA-256 protocols 
2. x.509 certificates 

Both KeyFiles and certificates are stored as files with minimal permissions. Note MongoDB disables support for TLS 1.0 encryption on systems where TLS 1.1+ is available.

These files need to be readable and writable by root (as these files are created by the charm), and readable by the `snap_daemon` user running the Charmed MongoDB snap commands.

### Client and router authentication to MongoDB

Clients can authenticate to MongoDB using:

1. username and password exchanged using SCRAM-SHA-256 protocols 
2. client certificates or CAs (mTLS)

When using SCRAM, usernames and passwords are stored in Charmed MongoDB and Charmed Mongos to be used by the mongod and mongos processes, in peer-relation data and in external relation to be shared with client applications via Juju secrets. 

When using mTLS, client certificates are loaded into a `tls-certificates` operator and provided to the Charmed MongoDB via a plain-text unencrypted 

> relation. Certificates are stored in a file with minimal permissions. Note MongoDB disables support for TLS 1.0 encryption on systems where TLS 1.1+ is available.

Reviewer: ```suggestion
relation. Certificates are stored in a file with minimal permissions. Note MongoDB disables support for TLS 1.0 and TLS 1.1 encryption.
```
